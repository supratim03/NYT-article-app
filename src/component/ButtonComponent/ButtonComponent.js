import React from "react";

const ButtonComponent = ({ classValue, handleSearch, buttonText, isPrev, disabledValue }) => {

	const handleClick = () => {
		handleSearch();
	};
	return (
		<button
			className={classValue}
			onClick={handleClick}
			disabled={isPrev ? disabledValue === 0 : disabledValue < 10}
		>
			<span>
				{buttonText}
			</span>
		</button>
	);
};

export default ButtonComponent;
