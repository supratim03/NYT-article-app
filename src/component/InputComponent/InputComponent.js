import React from "react";
import "../../styles/InputComponent.css";
const InputComponent = ({ handleInputChange, value, classValue, triggerSearch}) => {

	const handleInputOnChange = ({target: {value}}) => {
		handleInputChange(value);
	};
	const handleSearchAction = e => {
		if(e.keyCode === 13){
			triggerSearch();
		}
	};
	return (
		<input 
			className={classValue}
			placeholder="Search Article"
			onChange={handleInputOnChange}
			value={value}
			onKeyDown={handleSearchAction}
		/>
	);
};

export default InputComponent;
