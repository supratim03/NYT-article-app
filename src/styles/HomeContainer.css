.search-container-wrapper {
    display: flex;
    justify-content: center;
    height: 100vh;
    flex-direction: column;
    align-items: center;
    background: url("./../assets/Background.png");
}

.search-btn {
    background-color: #D5333E;
    border: 1px solid #D5333E;
    border-radius: 25px;
    color: #fff;
    font-size: 14px;
    cursor: pointer;
    padding: 15px 30px;
}

.search-btn:hover {
    box-shadow: 0 1px 1px rgb(0 0 0 / 10%);
    background-color: transparent;
    border: 1px solid #D5333E;
    color: #000000;
}

.wrapper {
    display: flex;
}

@media only screen and (max-width: 1000px) {
    .search-box {
        width: 300px !important;
    }



    .wrapper {
        flex-direction: column;
        justify-content: center;
        align-items: center;
    }

    .input-wrapper {
        padding-right: 0px !important;
    }
    
}

@media screen and (device-aspect-ratio: 40/71) {
    .search-input {
        width: 250px !important;
    }
}

.intro {
    content: url("./../assets/Intro.png");
    height: 150px;
    margin: auto;
}

.input-wrapper {
    padding-right: 25px;
}