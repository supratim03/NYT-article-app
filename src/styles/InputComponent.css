@media only screen and (max-width: 1000px) {
    .search-item-input {
        width: 85% !important;
    }
}

.search-input {
    width: 500px;
    border-radius: 25px;
    border: 1px solid #dfe1e5;
    box-shadow: none;
    padding: 17px 20px;
}

.search-input:hover {
    box-shadow: 0 1px 6px rgb(32 33 36 / 28%);
    border-color: rgba(223,225,229,0);
}

.search-input:focus-visible {
   outline: none;
}

@media screen and (device-aspect-ratio: 40/71) {
    .search-item-input {
        width: % !important;
    }
}