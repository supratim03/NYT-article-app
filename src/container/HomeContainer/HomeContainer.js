import React from "react";
import {useNavigate} from "react-router-dom";
import "../../styles/HomeContainer.css";
import InputComponent from "../../component/InputComponent/InputComponent";
import ButtonComponent from "../../component/ButtonComponent/ButtonComponent";

const HomeContainer = () => {
  const [inputVal, setInputVal] = React.useState("");
  const navigate = useNavigate();
  const handleInputChange = value => {
    setInputVal(value);
  };

  const handleSearch = () => {
    if(inputVal){
      navigate(`/search?query=${inputVal}`);
    }
  };
  return (
    <div className="search-container-wrapper">
      <div>
        <div className="intro"></div>
        <div className="logo sa-padding-top-20"></div>
      </div>
      <div className="wrapper">
        <div className="sa-padding-top-20 input-wrapper">
            <InputComponent
              handleInputChange={handleInputChange}
              value={inputVal}
              classValue="search-input search-box"
              triggerSearch={handleSearch}
            />
        </div>
        <div className="sa-padding-top-20">
            <ButtonComponent
              handleSearch={handleSearch}
              buttonText={"Search"}
              classValue="search-btn"
            />
        </div>
      </div>
    </div>
  );
};

export default HomeContainer;
