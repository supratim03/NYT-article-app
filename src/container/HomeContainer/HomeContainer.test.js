import React from "react";
import { mount } from "enzyme/build";
// import configureStore from "redux-mock-store";
// import createSagaMiddleware from "@redux-saga/core";
// import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import HomeContainer from "./HomeContainer";
import InputComponent from "../../component/InputComponent/InputComponent";
import ButtonComponent from "../../component/ButtonComponent/ButtonComponent";

// mock useNavigate
const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
   useNavigate: () => mockedUsedNavigate
}));

describe("Home container", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(
            <MemoryRouter>
                <HomeContainer />
            </MemoryRouter>
        );
    });

    it("render correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("Input on change" , () => {
        wrapper.find(InputComponent).invoke("handleInputChange")("USA");
        setTimeout(() => {
            expect(wrapper.text()).toEqual("USA");
        }, 0);
    });

    it("navigate new page" , () => {
        wrapper.find(InputComponent).invoke("handleInputChange")("USA");
        wrapper.find(ButtonComponent).invoke("handleSearch")();
        setTimeout(() => {
            expect(mockedUsedNavigate).toHaveBeenCalledTimes(1);
        }, 0);
    });
});

