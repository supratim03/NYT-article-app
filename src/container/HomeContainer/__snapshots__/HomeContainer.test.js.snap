// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home container render correctly 1`] = `ReactWrapper {}`;
